tensorboard/plugins/profile/pod_viewer/stack_bar_chart/stack-bar-chart.ts

```diff
@@ -114,53 +134,49 @@ Polymer({
             function(d) {
               d3.select(this).style('opacity', 1.0);
               parent.activeBar = null;
-            });
+            })
+        .transition()
+        .duration(TRANSITION_DURATION);
+    layer.exit().remove();
   },
   /**
    * Draw the axes of the chart.
    */
   drawAxes: function(svg: any, xScale: any, yScale: any, height: number) {
-    svg.append('g')
-        .attr('class', 'axis axis--x')
-        .style('font-size', FONT_SIZE)
-        .attr('transform', 'translate(0,' + (height + 5) + ')')
+    svg.select('.x.axis')
+        .transition()
+        .duration(TRANSITION_DURATION)     
         .call(d3.axisBottom(xScale));
-    svg.append('g')
-        .attr('class', 'axis axis--y')
-        .style('font-size', FONT_SIZE)
-        .attr('transform', 'translate(0,0)')
+    svg.select('.y.axis')
+        .transition()
+        .duration(TRANSITION_DURATION)
         .call(d3.axisLeft(yScale));
   },
   /**
    * Draw the legends of the chart.
    */
-  drawLegend: function(svg: any, labels: Array<string>, colorScale: any) {
-    let legend = svg.append('g')
-        .attr('font-family', 'sans-serif')
-        .attr('font-size', FONT_SIZE)
-        .attr('text-anchor', 'start')
-        .selectAll('g')
-        .data(labels.slice());
-    let legendG = legend.enter().append('g').merge(legend)
-        .attr('transform',
-            (d, i) => 'translate(' +
-                (i * LEGEND_WIDTH -
-                    Math.floor(i / LABELS_PER_LANE) * LEGEND_WIDTH *
-                        LABELS_PER_LANE) + ',' +
-                            Math.floor(i / LABELS_PER_LANE) *
-                                LEGEND_HEIGHT + ')'
-        );
+  drawLegend: function(selection: any, labels: Array<string>, colorScale: any) {
+    let legend = selection.selectAll('g').data(labels.slice());
+    legend.exit().remove();
 
-    legendG.append('rect')
-        .attr('x', YAXIS_TO_LEGEND)
+    let legendEnter = legend.enter().append('g');
+    legendEnter.append('rect')
         .attr('width', ICON_SIZE)
-        .attr('height', ICON_SIZE)
-        .attr('fill', (d, i) => colorScale(i));
-    legendG.append('text')
+        .attr('height', ICON_SIZE);
+    legendEnter.append('text')
         .attr('x', YAXIS_TO_LEGEND + LEGEND_MARGIN + ICON_SIZE)
         .attr('y', LEGEND_TEXT_HEIGHT)
         .attr('dy', LEGEND_TEXT_SIZE)
-        .text((d) => d);
+   
+    legend = legendEnter.merge(legend);
+    legend.attr('transform',
+        (d, i) => 'translate(' + (i * LEGEND_WIDTH -
```

Comment:
this statement is rather hard to read. Can you please assign math to more semantic variable? i.e.,
```js
const x = i * LEGEND - Math.floor(i / LABELS_PER_LANE) * LEGEND_WIDTH *
                LABELS_PER_LANE;
const y = ...
return `translate(${x}, ${y})`;

```

Reply:
Yep, your suggestion is much cleaner. Done.